include/devicetree/clocks.h

```diff
@@ -54,6 +54,39 @@ extern "C" {
 #define DT_CLOCKS_LABEL_BY_IDX(node_id, idx) \
 	DT_PROP_BY_PHANDLE_IDX(node_id, clocks, idx, label)
 
+/**
+ * @brief Get clock controller "name" (label property) by name
+ *
+ * It's an error if the clock controller referenced by the phandle
+ * in property "clocks" at index "idx" has no label property.
+ *
+ * Example devicetree fragment:
+ *
+ *     clk1: clkctrl@... {
```

Comment:
I took a look, and it seems like dtspec recommends either 'clock' or 'clock-controller' for these. Think it's worth changing the examples to align with the spec as follow up work? I know this is just using the existing convention so it's fine for me to keep it as is for this PR.

Reply:
Added a commit in this PR for it.